src/pseudopeople/interface.py

```diff
@@ -64,7 +65,16 @@ def _generate_form(
             f"Source {source} must be either a pandas DataFrame or a path to a "
             "file containing a pandas DataFrame."
         )
-    return noise_form(form, data, configuration_tree, seed)
+
```

Comment:
What is the logic for applying dtypes before noising, where much of the noising will change it all to strings and then maybe not back to the initial dtype?

Reply:
@rmudambi ^

Reply:
The reason is that in post-processing we converted all columns that are supposed to be strings into categorical for savings in both space and file read-time. We don't want them to be categorical anymore, since we're noising them as strings.

Reply:
We should have tests that verify that the data post noising has the expected dtypes.